src/Gizmos/gizmoManager.ts

```diff
@@ -136,32 +139,34 @@ export class GizmoManager implements IDisposable {
             }
             if (pointerInfo.type == PointerEventTypes.POINTERDOWN) {
                 if (pointerInfo.pickInfo && pointerInfo.pickInfo.pickedMesh) {
-                    var node: Nullable<Node> = pointerInfo.pickInfo.pickedMesh;
-                    if (this.attachableMeshes == null) {
-                        // Attach to the most parent node
-                        while (node && node.parent != null) {
-                            node = node.parent;
-                        }
-                    } else {
-                        // Attach to the parent node that is an attachableMesh
-                        var found = false;
-                        this.attachableMeshes.forEach((mesh) => {
-                            if (node && (node == mesh || node.isDescendantOf(mesh))) {
-                                node = mesh;
-                                found = true;
+                    if (this.enableAutoPicking) {
+                        var node: Nullable<Node> = pointerInfo.pickInfo.pickedMesh;
+                        if (this.attachableMeshes == null) {
+                            // Attach to the most parent node
+                            while (node && node.parent != null) {
+                                node = node.parent;
+                            }
+                        } else {
+                            // Attach to the parent node that is an attachableMesh
+                            var found = false;
+                            this.attachableMeshes.forEach((mesh) => {
```

Comment:
you can use `.some` instead of forEach, and break when you found it. the variable `found` will get the result of the `.some` call.

Reply:
This code has only been indented for the if. I don't want to do any change that may cause more problem.